dask_xgboost/core.py

```diff
@@ -82,12 +83,25 @@ def train_part(env, param, list_of_parts, dmatrix_kwargs=None, **kwargs):
     dmatrix_kwargs["feature_names"] = getattr(data, 'columns', None)
     dtrain = xgb.DMatrix(data, labels, **dmatrix_kwargs)
 
+    if eval_set is not None:
+        if sample_weight_eval_set is None:
+            sample_weight_eval_set = [None] * len(eval_set)
+        evals = list(
+            xgb.DMatrix(eval_set[i][0].compute(), label=eval_set[i][1].compute(),
```

Comment:
It is odd to call compute from within a task.  It's doable, but rarely optimal.  This gets to the question below about supportind dask arrays